### PR TITLE
Feature/infinite query helpers

### LINKED ISSUE #1358: Add `setInfiniteQueryData` helper
if you have `useInifiteQuery(['foo', {}])` then `args` in `setQueryData<..., TInput, ...>(['foo', {}], args => ...`  is incorrectly typed. Types say that `args` is `TInput` where the actual value of `args` if inspected is `{ pageParams: ...,  pages: TInput[] }`

### PR DESCRIPTION
**Pull request by BrockHerion**
Closes #1358 

Changes:
- Added a `setInfiniteQueryData` helper that correctly types input to `InfiniteData`
- Added a `getInfiniteQuertData` helper that correctly types output to `InfiniteData`
- Created test for `setInfiniteQueryData`
- Updated docs page to include both helps with examples

### CODE REVIEW

**Comment by KATT** [comment]
@all-contributors ass @BrockHerion for code, test & docs

**Comment by KATT** [comment]
@all-contributors add @BrockHerion for code, test & docs

File: packages/react/src/internals/context.tsx [hunk 1]
```diff
@@ -176,6 +176,28 @@ export interface TRPCContextState<
...
+    TOutput extends inferProcedureOutput<TRouter['_def']['queries'][TPath]>,
+  >(
+    pathAndInput: [TPath, TInput?],
+    updater: Updater<InfiniteData<TOutput> | undefined, InfiniteData<TOutput>>,
+    options?: SetDataOptions,
+  ): void;
+  /**
+   * @link https://react-query.tanstack.com/reference/QueryClient#queryclientgetquerydata
+   */
+  getInifiniteQueryData<
```
Comment (sachinraja) [inline packages/react/src/internals/context.tsx:194]:
```suggestion
  getInfiniteQueryData
```
Spelling mistake

File: www/docs/react/useInfiniteQuery.md [hunk 2]
```diff
@@ -76,4 +76,67 @@ function MyComponent() {
...
+import { trpc } from '../utils/trpc';
+
+function MyComponent() {
+  const utils = trpc.useContext();
+
+  const myMutation = trpc.useMutation('infinitePosts.delete', {
+    onMutate({ post }) {
+      await utils.cancelQuery(['infinitePosts']);
+
+      utils.setInfiniteQueryData(['infinitePosts', { limit: 10 }] (data) => {
```
Comment (KATT) [inline www/docs/react/useInfiniteQuery.md:119]:
```suggestion
      utils.setInfiniteQueryData(['infinitePosts', { limit: 10 }], (data) => {
```

**Review by KATT — APPROVED** [review]
lgtm!

**Review by KATT — APPROVED** [review]

**Pull request merged**